uniffi_bindgen/src/bindings/kotlin/gen_kotlin/mod.rs

```diff
@@ -569,31 +584,54 @@ mod filters {
     }
 
     /// Get the idiomatic Kotlin rendering of a function name.
-    pub fn class_name(nm: &str, ci: &ComponentInterface) -> Result<String, askama::Error> {
+    pub fn class_name(
+        nm: &str,
+        ci: &ComponentInterface,
+        config: &Config,
+    ) -> Result<String, askama::Error> {
+        if let Some(overwrite_class_name) = config.rename.get(nm) {
+            return Ok(KotlinCodeOracle.class_name(ci, overwrite_class_name));
+        }
         Ok(KotlinCodeOracle.class_name(ci, nm))
     }
 
     /// Get the idiomatic Kotlin rendering of a function name.
-    pub fn fn_name(nm: &str) -> Result<String, askama::Error> {
+    pub fn fn_name(nm: &str, config: &Config) -> Result<String, askama::Error> {
+        if let Some(overwrite_fn_name) = config.rename.get(nm) {
+            return Ok(KotlinCodeOracle.fn_name(overwrite_fn_name));
+        }
         Ok(KotlinCodeOracle.fn_name(nm))
     }
 
     /// Get the idiomatic Kotlin rendering of a variable name.
-    pub fn var_name(nm: &str) -> Result<String, askama::Error> {
+    pub fn var_name(nm: &str, config: &Config) -> Result<String, askama::Error> {
+        if let Some(overwrite_var_name) = config.rename.get(nm) {
+            return Ok(KotlinCodeOracle.var_name(overwrite_var_name));
+        }
```

Comment:
Oh, through all the debugging I totally forgot my note on changing this to `[type_name].[field_name]`. Thanks for reminding me! 

So we put a lid on this for now and focus on a cleaner structure first (based on your opened issue) or should I still adjust to `[type_name].[field_name]`? 

It was very confusing for me what `var_name` changes. So yes, totally agree.

Reply:
I will push a change to adopt `[type_name].[field_name]`. Just to double check since I don't have a gut feeling yet about how people use this feature: We expect `type_name` to be the `type` of the Rust representation, or the associated type of the target language?

Reply:
I think the Rust representation is best.  I think we've used that in other cases when we want to identify things.  Also, it makes cut+paste easier when you want to rename something in multiple languages.